### PR TITLE
gooddata-js RELATED: BB-1780 Measure value filters in AFM filter merging function

### PR DESCRIPTION
**Pull request by xMort**
---

# PR checklist

- [ ] Change was tested using dev-release in Analytical Designer and Dashboards (if applicable).
- [ ] Change is described in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [ ] Migration guide (for major update) is written to [CHANGELOG.md](../blob/master/CHANGELOG.md).

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-oss-check/21/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/446/ : SUCCESS in 3m 44s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-oss-check/22/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/447/ : SUCCESS in 1m 55s

File: src/DataLayer/utils/AfmUtils.ts [hunk 1]
```diff
@@ -321,9 +323,14 @@ export function appendFilters(
...
 
-    const afmAttributeFilters = afmFilters.filter(filter => !AFM.isDateFilter(filter));
+    const afmNonDateFilters = afmFilters.filter(filter => !AFM.isDateFilter(filter));
 
-    const filters = compact([...afmAttributeFilters, ...attributeFilters, ...dateFilters]);
+    const filters = compact([
+        ...afmNonDateFilters,
+        ...attributeFilters,
+        ...dateFilters,
+        ...(measureValueFilters || []),
```
Comment (tomassatinsky) [inline src/DataLayer/utils/AfmUtils.ts]: Isn't the `compact` function handling the undefined value so the `|| []` is not necessary?

Comment (xMort) [reply src/DataLayer/utils/AfmUtils.ts]: It is, otherwise spread operator on possibly undefined value would throw an error.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-oss-check/23/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/448/ : SUCCESS in 1m 52s

File: src/DataLayer/utils/AfmUtils.ts [hunk 2]
```diff
@@ -321,9 +323,14 @@ export function appendFilters(
         dateFilters.push(afmDateFilter);
     }
 
-    const afmAttributeFilters = afmFilters.filter(filter => !AFM.isDateFilter(filter));
+    const afmNonDateFilters = afmFilters.filter(filter => !AFM.isDateFilter(filter));
```
Comment (tomassatinsky) [inline src/DataLayer/utils/AfmUtils.ts]: the structure of this function is terrible/unreadable. I would appreciate some improvement but if you don't have time for that we can merge as this PR doesn't make it any worse.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-oss-check/24/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/449/ : SUCCESS in 1m 54s

File: src/DataLayer/utils/AfmUtils.ts (hunk 2 above)
Comment (xMort) [reply src/DataLayer/utils/AfmUtils.ts]: It is not great, I agree with that.
However, I would leave it for now. I would rather work on the next task and then help with sprint goals.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-js-merge-compliance-check/102/ : SUCCESS in 21s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-js-unit-tests-zuul-docker-el7/450/ : SUCCESS in 57s

**Pull request merged**